lib/api/oauth.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Paytm.API.OAuth do
+  use HTTPoison.Base
+
+  @config        Application.get_env(:paytm, Paytm.API.OAuth)
+  @base_url      @config[:base_url]
+  @client_id     @config[:client_id]
+  @client_secret @config[:client_secret]
+
+  def process_url(url) do
+    @base_url <> url
+  end
+
+  @send_otp_error_codes %{
+    "430" => :invalid_authorization,
+    "431" => :invalid_mobile,
+    "432" => :login_failed,
+    "433" => :account_blocked,
+    "434" => :bad_request,
+    "465" => :invalid_email
+  }
+
+  @spec send_otp(%{email: String.t, phone: String.t}) :: {:ok, state :: String.t, :login}
+                                                      |  {:ok, state :: String.t, :register}
+                                                      |  {:error, message :: String.t | atom, code :: atom}
+  def send_otp(%{email: email}) when is_nil(email) or email == "", do: {:error, "`email` is required", nil}
+  def send_otp(%{phone: phone}) when is_nil(phone) or phone == "", do: {:error, "`phone` is required", nil}
+  def send_otp(%{email: email}) when not is_binary(email), do: {:error, "`email` must be a string", nil}
+  def send_otp(%{phone: phone}) when not is_binary(phone), do: {:error, "`phone` must be a string", nil}
```

Comment:
I think you can do `when field == "" or not is_binary(field)`
The `not is_binary` should take care of nil values